backend/controllers/controller_kubernetes_resources.go

```diff
@@ -3,110 +3,108 @@ package controllers
-		return
-	}
-
-	w.WriteHeader(http.StatusOK)
-
-	status := models.Status{
-		Status:  "Success",
-		Code:    http.StatusOK,
-		Message: fmt.Sprintf("Resource %s deleted successfully", resourceName),
-	}
-	writeJSONResponse(w, http.StatusOK, status)
+	handleResourceOperation(w, r, models.Delete, func(resourceType, namespace, resourceName string) (interface{}, *models.ModelError) {
+		if err := cluster.DeleteResource(resourceType, namespace, resourceName); err != nil {
+			return nil, err
+		}
+		return models.Status{
+			Status:  "Success",
+			Code:    http.StatusOK,
+			Message: fmt.Sprintf("Resource %s deleted successfully", resourceName),
+		}, nil
+	})
 }
 
 func UpdateResourceController(w http.ResponseWriter, r *http.Request) {
-	setJSONContentType(w)
+	handleResourceOperation(w, r, models.Update, func(resourceType, namespace, resourceName string) (interface{}, *models.ModelError) {
+		var resource models.ResourceDetails
+		if !decodeJSONBody(w, r, &resource.ResourceDetails) {
+			return nil, &models.ModelError{Code: http.StatusBadRequest, Message: "Invalid request body"}
+		}
+		return cluster.UpdateResource(resourceType, namespace, resourceName, resource)
+	})
+}
 
+func handleResourceOperation(w http.ResponseWriter, r *http.Request, opType models.OperationType, operationFunc func(string, string, string) (interface{}, *models.ModelError)) {
 	resourceType := getResourceType(r)
 	resourceName := getResourceName(r)
 	namespace := getNamespace(r)
 
-	var resource models.ResourceDetails
-	if !decodeJSONBody(w, r, &resource.ResourceDetails) {
+	operation := models.Operation{
+		Resource:  resourceType,
+		Namespace: namespace,
+		Type:      opType,
+	}
+
+	if err := setJSONCTAndAuth(w, r, operation); err != nil {
+		writeJSONResponse(w, int(err.Code), err)
 		return
 	}
 
-	resource, err := cluster.UpdateResource(resourceType, namespace, resourceName, resource)
+	result, err := operationFunc(resourceType, namespace, resourceName)
 	if err != nil {
 		writeJSONResponse(w, int(err.Code), err)
 		return
 	}
 
-	writeJSONResponse(w, http.StatusOK, resource)
+	statusCode := http.StatusOK
+	if opType == models.Create {
+		statusCode = http.StatusCreated
+	}
+
+	writeJSONResponse(w, statusCode, result)
 }
 
-func setJSONContentType(w http.ResponseWriter) {
+func setJSONCTAndAuth(w http.ResponseWriter, r *http.Request, operation models.Operation) *models.ModelError {
 	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	authErrors := map[string]*models.ModelError{
+		"Authentication failed": {
+			Code:    http.StatusUnauthorized,
+			Message: "Authentication failed",
+		},
+		"Insufficient permissions": {
+			Code:    http.StatusForbidden,
+			Message: "Insufficient permissions",
+		},
+	}
+	token, err := auth.GetJWTTokenFromHeader(r)
+	if err != nil || !auth.IsTokenValid(token) {
+		return authErrors["Authentication failed"]
+	}
+	if !auth.IsUserAuthorized(operation) {
+		return authErrors["Insufficient permissions"]
+	}
```

Comment:
I think this is more clean solution and doesn't risk making a typo in getting value from the map which can result in returning nil instead of error
```suggestion
	token, err := auth.GetJWTTokenFromHeader(r)
	if err != nil || !auth.IsTokenValid(token) {
		return &models.ModelError{
            Code:    http.StatusUnauthorized,
			Message: "Authentication failed",
        }
	}
	if !auth.IsUserAuthorized(operation) {
		return &models.ModelError{
            Code:    http.StatusForbidden,
			Message: "Insufficient permissions",
        }
	}
```

Reply:
Done